### PR TITLE
Apply new language features

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Mostly done by solution-wide auto fix in IDE. Per feature per commit, suggesting to review by commit.

Features not enabled:
- ternary operator
  not distinguishable with switch expressions
- coalesce throw expressions
  probably succeeded by C# 9 champion "simplified parameter null-checking", which has started
- `ReferenceEquals(null)`
  covered by other PR
- switch expression
  waiting for ReSharper fix
- index/range and await using
  missing corresponding member on `netstandard2.0`
- nullable reference types
  should be an epic
- others that no IDE suggestions now
  like pattern matching with nested properties

### CODE REVIEW

File: osu-framework.sln.DotSettings [hunk 1]
```diff
@@ -118,6 +118,7 @@
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ParameterHidesMember/@EntryIndexedValue">HINT</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ParameterOnlyUsedForPreconditionCheck_002EGlobal/@EntryIndexedValue">HINT</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ParameterOnlyUsedForPreconditionCheck_002ELocal/@EntryIndexedValue">HINT</s:String>
+	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=PatternAlwaysOfType/@EntryIndexedValue">DO_NOT_SHOW</s:String>
```
Comment (huoyaoyuan) [inline osu-framework.sln.DotSettings]: ReSharper suggests changing `if (x is TStruct t)` where `TStruct? x` to `if (x is { } t)`, but it then infers wrong type of `{ } t`. Moreover, I think "`{ }` means not null" is really not straightforward. Likely to disable forever.

File: osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs [hunk 2]
```diff
@@ -19,23 +19,21 @@ public AndroidTextureLoaderStore(IResourceStore<byte[]> store)
...
-                byte[] result = new byte[pixels.Length * sizeof(int)];
-                Buffer.BlockCopy(pixels, 0, result, 0, result.Length);
-
-                for (int i = 0; i < pixels.Length; i++)
-                {
-                    var b = result[i * 4];
-                    result[i * 4] = result[i * 4 + 2];
-                    result[i * 4 + 2] = b;
-                }
+            using var bitmap = BitmapFactory.DecodeStream(stream);
```
Comment (peppy) [inline osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs]: not a huge fan.

Comment (bdach) [reply osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs]: Yeah, I also don't like this. Having explicit scoping in `using` blocks is a feature for me, swapping them for implicit scoping like this makes everything harder to read. I had to stop and pause for like 5 seconds to understand why this isn't broken.

Comment (huoyaoyuan) [reply osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs]: The IDE suggestion currently only applies when the scope lives to end of parent scope.
I'm not a hige fan too. Going to make this optional.

Comment (smoogipoo) [reply osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs]: Would like to see all changes relating to this reversed in this PR.

Comment (huoyaoyuan) [reply osu.Framework.Android/Graphics/Textures/AndroidTextureLoaderStore.cs]: Reverted. Not literally "everything", not including IDE settings to not suggest it.

**Pull request merged**